src/go/wsl-helper/pkg/host/hostfile.go

```diff
@@ -0,0 +1,158 @@
+	EndConfig           = "# END Rancher Desktop configuration."
+	DefaultHostFilePath = "/etc/hosts"
+)
+
+// AppendHostsFile reads the content of a host file
+// and appends a slice of entries to the file. The entries are
+// provided in the following format, e.g.:
+//
+//	newEntries := []string{
+//	  "127.0.0.1,example.com",
+//	  "127.0.0.1,another-example.com",
+//	}
+func AppendHostsFile(entries []string, hostsFilePath string) error {
+	exist, err := configExist(hostsFilePath)
+	if err != nil {
+		return err
+	}
+	if exist {
+		return nil
+	}
+
+	hostFile, err := os.OpenFile(hostsFilePath, os.O_WRONLY|os.O_APPEND, 0644)
+	if err != nil {
+		return err
+	}
+	defer hostFile.Close()
+
+	writer := bufio.NewWriter(hostFile)
+
+	_, err = writer.WriteString(fmt.Sprintf("\n%s\n", BeginConfig))
+	if err != nil {
+		return err
+	}
+	for _, entry := range entries {
+		_, err := writer.WriteString(strings.ReplaceAll(entry, ",", " ") + "\n")
+		if err != nil {
+			return fmt.Errorf("error writing to /etc/hosts file: %w", err)
+		}
+	}
+	_, err = writer.WriteString(fmt.Sprintln(EndConfig))
+	if err != nil {
+		return err
+	}
+
+	if err = writer.Flush(); err != nil {
+		return fmt.Errorf("error flushing writer: %w", err)
+	}
+
+	return nil
+}
+
+// RemoveHostsFileEntry reads the content of a host file
+// and removes any specific host declaration that belongs
+// to Rancher Desktop.
+func RemoveHostsFileEntry(hostsFilePath string) error {
+	exist, err := configExist(hostsFilePath)
+	if err != nil {
+		return err
+	}
+	if !exist {
+		return nil
+	}
+
+	hostsFile, err := os.Open(hostsFilePath)
+	if err != nil {
+		return err
+	}
+	defer hostsFile.Close()
+
+	tempFile, err := os.CreateTemp("", "tempfile")
+	if err != nil {
+		return err
+	}
+	defer tempFile.Close()
+
+	insideRancherConfig := false
+
+	scanner := bufio.NewScanner(hostsFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// Check if we are inside Rancher Desktop configuration
+		if strings.Contains(line, BeginConfig) {
+			insideRancherConfig = true
+			continue
+		} else if strings.Contains(line, EndConfig) {
+			insideRancherConfig = false
+			continue
+		}
+
+		// Keep lines that are not inside the Rancher Desktop configuration
+		if !insideRancherConfig {
+			_, err := tempFile.WriteString(line + "\n")
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return err
+	}
+
+	if err := tempFile.Close(); err != nil {
+		return err
+	}
+
+	// Replace the original file with the temporary file
+	if err := os.Rename(tempFile.Name(), hostsFilePath); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func configExist(hostsFilePath string) (bool, error) {
+	hostFile, err := os.OpenFile(hostsFilePath, os.O_RDWR, 0644)
+	if err != nil {
+		return false, err
+	}
+	defer hostFile.Close()
+	scanner := bufio.NewScanner(hostFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, BeginConfig) {
```

Comment:
It's better to check for `EndConfig` in case we have it half-written, I think. (Even better to check for both, in the expected order, but meh)